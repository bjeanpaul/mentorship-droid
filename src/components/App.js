import React from 'react-native';
import { Provider } from 'react-redux';

import configureStore from '../stores/configureStore';
const store = configureStore();


console.log('----------------------------')
import { login } from '../mentor/MentorActions';
store.dispatch(login('admin', '123'));
console.log('----------------------------')


import RootNavigator from './RootNavigator';



const App = function App() {
  return (
    <Provider store={store}>
        <RootNavigator />
    </Provider>
  );
};

export default App;
