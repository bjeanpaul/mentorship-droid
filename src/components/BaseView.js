import React from 'react';
import { View, StyleSheet, StatusBar } from 'react-native';
import colors from 'src/constants/colors';


const styles = StyleSheet.create({
  view: {
    backgroundColor: colors.DEFAULT_BG,
    flex: 1,
  },
});

const BaseView = ({
  children,
  style,
  statusBarBackgroundColor = colors.DEFAULT_STATUS_BG,
}) => (
  <View style={[styles.view, style]}>
    { statusBarBackgroundColor
      ? <StatusBar backgroundColor={statusBarBackgroundColor} />
      : null
    }
    {children}
  </View>
);
BaseView.propTypes = {
  statusBarBackgroundColor: React.PropTypes.string,
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};
export default BaseView;
