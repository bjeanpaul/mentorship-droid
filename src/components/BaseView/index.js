import React from 'react';
import { View } from 'react-native';

import styles from './styles';


const BaseView = ({
  children,
  style,
}) => (
  <View style={[styles.view, style]}>
    {children}
  </View>
);


BaseView.propTypes = {
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};


export default BaseView;
