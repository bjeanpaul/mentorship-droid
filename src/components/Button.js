import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View, StyleSheet } from 'react-native';

import Text from 'Text';
import { font, fontWeight } from 'src/app/styles';


const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f77040',
    borderRadius: 5,
    marginLeft: 16,
    marginRight: 16,
  },
  buttonDisabled: {
    backgroundColor: '#dfe5e6',
  },
  label: {
    color: '#fff',
    fontFamily: font.brandonTextBold,
    fontSize: 13,
    fontWeight: fontWeight.bold,
    letterSpacing: 1,
    marginTop: 15,
    marginBottom: 17,
  },
});

const Button = ({
  label,
  disabled,
  handlePress,
}) => {
  let button = (
    <View style={[styles.button, disabled && styles.buttonDisabled]}>
      <Text style={styles.label}>{label.toUpperCase()}</Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={handlePress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};
Button.propTypes = {
  label: PropTypes.string.isRequired,
  handlePress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};
export default Button;
