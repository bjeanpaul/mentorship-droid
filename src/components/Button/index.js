import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View } from 'react-native';
import Text from 'src/components/Text';
import styles, { themes, layouts } from './styles';


const Button = ({
  theme = themes.default,
  layout = layouts.default,
  children,
  disabled,
  onPress,
}) => {
  let button = (
    <View
      style={[
        styles.container,
        theme.container,
        layout,
        disabled && theme.containerIsDisabled,
      ]}
    >
      <Text
        style={[
          styles.title,
          theme.title,
          disabled && theme.titleIsDisabled,
        ]}
      >
        {children}
      </Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={onPress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};


Button.themes = themes;
Button.layouts = layouts;


Button.propTypes = {
  children: PropTypes.any,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  theme: PropTypes.any,
  layout: PropTypes.any,
};


export default Button;
