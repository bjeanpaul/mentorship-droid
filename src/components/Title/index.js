import React, { PropTypes } from 'react';
import styles, { themes } from './styles';
import Text from 'src/components/Text';


export const Title = ({
  style = {},
  children = '',
  theme = {},
}) => (
  <Text style={[styles.title, theme.title, style]}>
    {children}
  </Text>
);


Title.propTypes = {
  style: PropTypes.any,
  children: PropTypes.string,
  theme: PropTypes.object,
};


Title.themes = themes;


export default Title;
