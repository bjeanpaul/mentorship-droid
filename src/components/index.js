import Text from './Text';
import Label from './Label';
import Link from './Link';
import Button from './Button';
import TextInput from './TextInput';
import MultiLineTextInput from './MultiLineTextInput';
import BaseView from './BaseView';
import Toolbar from './Toolbar';
import Header from './Header';
import Title from './Title';
import Icon from './Icon';
import HeaderIcon from './HeaderIcon';
import OverlayCompleted from './Overlay/Completed';
import OverlayLoading from './Overlay/Loading';
import Section from './Section';

export {
  BaseView,
  Toolbar,
  Text,
  Label,
  Link,
  Button,
  TextInput,
  MultiLineTextInput,
  Toolbar,
  Header,
  Title,
  Icon,
  HeaderIcon,
  OverlayCompleted,
  OverlayLoading,
  Section,
};
