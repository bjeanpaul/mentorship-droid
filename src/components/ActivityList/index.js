import React, { PropTypes } from 'react';
import { View, Image, ScrollView, TouchableNativeFeedback } from 'react-native';
import { Text } from 'src/components';
import images from 'src/constants/images';
import { ImageUrl } from 'src/api';
import styles from './styles';


const ActivityList = ({
  category: { color },
  activities,
  onActivityPress,
}) => (
  <ScrollView>
    {activities.map(({ id, title, icon, isComplete }) => (
    <TouchableNativeFeedback
      key={id}
      uid={`activity:${id}`}
      onPress={() => onActivityPress(id)}
    >
      <View style={styles.activity}>
        <ActivityIcon icon={icon} color={color} isComplete={isComplete} />
        <Text numberOfLines={3} style={styles.activityTitle}>{title}</Text>
      </View>
    </TouchableNativeFeedback>
    ))}
  </ScrollView>
);


const ActivityIcon = ({
  icon,
  color,
  isComplete,
}) => (
  <View style={styles.activityIconContainer}>
    <View
      style={[
        styles.activityIcon,
        isComplete
          ? { backgroundColor: color }
          : styles.activityIconIsIncomplete,
      ]}
    >
      <Image
        source={icon.toSource(images.ACTIVITY_ICON_FALLBACK)}
        style={styles.activityIconImage}
      />
    </View>

    {
      isComplete
        ? <Image
          style={styles.activityCompleteAnnotation}
          source={images.ACTIVITY_COMPLETE_ANNOTATION}
        />
        : null
    }
  </View>
);


ActivityList.propTypes = {
  category: PropTypes.object.isRequired,
  activities: PropTypes.array.isRequired,
  onActivityPress: PropTypes.func.isRequired,
};


ActivityIcon.propTypes = {
  icon: PropTypes.instanceOf(ImageUrl),
  color: PropTypes.string.isRequired,
  isComplete: PropTypes.bool.isRequired,
};


export default ActivityList;
