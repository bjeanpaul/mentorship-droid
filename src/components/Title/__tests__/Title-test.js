import React from 'react';
import { Title } from 'src/components';


describe('Title', () => {
  it('should render', () => {
    expect(render(
      <Title>Title</Title>
    )).toMatchSnapshot();
  });

  it('should support style overriding', () => {
    expect(render(
      <Title style={{ color: '#222' }}>Title with style</Title>
    )).toMatchSnapshot();
  });

  it('should support theming', () => {
    expect(render(
      <View style={{ backgroundColor: '#000' }}>
        <Title theme={Title.themes.light}>Title with theme</Title>
      </View>
    )).toMatchSnapshot();
  });
});
