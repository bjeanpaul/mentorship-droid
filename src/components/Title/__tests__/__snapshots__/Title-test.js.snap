exports[`Title should render 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center"
      },
      Array [
        Object {
          "fontFamily": "Brandon Text_medium",
          "fontSize": 20,
          "fontWeight": "400"
        },
        undefined,
        Object {}
      ]
    ]
  }>
  Microscopic Hummingbirds
</Text>
`;

exports[`Title should support style overriding 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center"
      },
      Array [
        Object {
          "fontFamily": "Brandon Text_medium",
          "fontSize": 20,
          "fontWeight": "400"
        },
        undefined,
        Object {
          "fontSize": 22
        }
      ]
    ]
  }>
  Microscopic Hummingbirds
</Text>
`;

exports[`Title should support theming 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center"
      },
      Array [
        Object {
          "fontFamily": "Brandon Text_medium",
          "fontSize": 20,
          "fontWeight": "400"
        },
        Object {
          "color": "#eee"
        },
        Object {}
      ]
    ]
  }>
  Microscopic Hummingbirds
</Text>
`;
